py/desisim/survey_release.py

```diff
@@ -0,0 +1,302 @@
+            zbins = np.arange(0,10.1,0.1)
+            zcenters=0.5*(zbins[1:]+zbins[:-1])
+            dndz=factor*np.interp(zcenters,dist['z'],dist['dndz_23'],left=0,right=0)
+        elif distribution=='from_data':
+            raise NotImplementedError(f"Option {distribution} is not implemented yet")
+        
+            if self.data is None:
+                raise ValueError("No data catalog was provided")
+            dz = 0.1
+            zbins = np.linspace(zmin,zmax,int((zmax-zmin)/dz))
+            data_area = SurveyRelease.get_catalog_area(self.data,nside=256)
+            zcenters=0.5*(zbins[1:]+zbins[:-1])
+            dndz=np.histogram(self.data['Z'],bins=zbins,weights=np.repeat(1/data_area,len(self.data)))[0]
+        else:
+            raise ValueError(f"Distribution option {distribution} not in available options: SV, target_selection, from_data")
+        mock_area=SurveyRelease.get_catalog_area(self.mockcatalog,nside=16)
+        dndz_mock=np.histogram(self.mockcatalog['Z_QSO_RSD'],bins=zbins,weights=np.repeat(1/mock_area,len(self.mockcatalog)))[0]
+        fractions=np.divide(dndz,dndz_mock,where=dndz_mock>0,out=np.zeros_like(dndz_mock))
+        fractions[fractions>1]=1
+
+        local_z = self.mockcatalog['Z_QSO_RSD']
+        bin_index = np.digitize(local_z, zbins) - 1
+        rand = np.random.uniform(size=local_z.size)
+        if self.invert:
+            rand = 1-rand
+        selection = rand < fractions[bin_index]
+        log.info(f"Keeping {sum(selection)} mock QSOs to match {distribution} distribution")
+        self.mockcatalog=self.mockcatalog[selection]
+        
+    def apply_data_geometry(self, release='iron'):
+        """ Apply the geometry of a given data release to the mock catalog.
+        
+        Args:
+            data (astropy.table.Table): data catalog.
+            release (str,optional): release name. If None, will use the whole DESI footprint.
+        """
+        mock=self.mockcatalog
+        tiles=self.get_lya_tiles(release)
+        mask_footprint=desimodel.footprint.is_point_in_desi(tiles,mock['RA'],mock['DEC'])
+        mock=mock[mask_footprint]
+        log.info(f"Keeping {sum(mask_footprint)}  mock QSOs in footprint TILES")
+        if release is not None:
+            # TODO: THis only works for iron at the moment.
+            # TODO 2: Add submoodules to generate the pixelmaps from data.
+            log.info(f"Downsampling by NPASSES fraction in {release} release")
+            # TODO: Implement this in desimodel instead of local path
+            pixmap=Table.read('/global/cfs/cdirs/desi/users/hiramk/desi/quickquasars/sampling_tests/npass_pixmap.fits')
+            mock_pixels = hp.ang2pix(1024, np.radians(90-mock['DEC']),np.radians(mock['RA']),nest=True)
+            try:
+                data_pixels = hp.ang2pix(1024,np.radians(90-self.data['TARGET_DEC']),np.radians(self.data['TARGET_RA']),nest=True)
+            except: 
```

Comment:
Not a good use of try-except. I recommend `if 'TARGET_DEC` in self.data.dtype.names` or similar

Reply:
Also, never use naked except!